### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by oleglukyanov**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class UsersController < ApplicationController
+
+  def show
+    @users = User.all
+    p "is this working?"
```
Comment (dearshrewdwit) [inline app/controllers/users_controller.rb]: Please remove debugging statements

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -23,6 +23,10 @@
   <body>
     <%= link_to 'Sign out', destroy_user_session_path, :method=>'delete' %>
 
+    <% if controller.current_user %>
```
Comment (dearshrewdwit) [inline app/views/layouts/application.html.erb]: Do you need the `controller` object here?

File: config/routes.rb [hunk 3]
```diff
@@ -1,10 +1,11 @@
 Rails.application.routes.draw do
+  get 'users/show'
+
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   # root 'posts'
-
   devise_for :users
   resources :posts
-
+  resources :users
```
Comment (dearshrewdwit) [inline config/routes.rb]: let's nest the resource like
http://guides.rubyonrails.org/routing.html#nested-resources
https://stackoverflow.com/questions/27712947/nested-resources-in-devise

Comment (dearshrewdwit) [reply config/routes.rb]:
```ruby
resources :users do
  resources :posts 
end
```

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+# RSpec.describe UsersController, type: :controller do
+#
+#   describe "GET #show" do
+#     it "returns http success" do
```
Comment (dearshrewdwit) [inline spec/controllers/users_controller_spec.rb]: if you need to not run a test, mark it as pending
https://relishapp.com/rspec/rspec-core/v/3-6/docs/pending-and-skipped-examples/pending-examples

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -18,6 +18,21 @@
 # See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
 
 RSpec.configure do |config|
+
+  config.before(:suite) do
```
Comment (dearshrewdwit) [inline spec/spec_helper.rb:22]: nice addition

File: Gemfile.lock [hunk 6]
```diff
@@ -206,6 +207,7 @@ PLATFORMS
 DEPENDENCIES
   byebug
   capybara (~> 2.13)
+  database_cleaner
```
Comment (dearshrewdwit) [inline Gemfile.lock:221]: 👍

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

**Comment by tobold** [comment]
Which ticket / user story is this branch for? Is anyone working on making requested changes?

**Comment by annalaise** [comment]
@dearshrewdwit @tobold all issues now resolved, apart from nested resources for posts and users. We need to do that once we understand how posts are connected to users - which happens once Users are set up after this merge, and all mocks are removed.

**Comment by annalaise** [comment]
@dearshrewdwit this is now all sorted. Review plez!

**Review by dearshrewdwit — APPROVED** [review]

**Pull request merged**